### DISCUSSION TITLE
Microsoft Word opening wrong file when double clicking on a .docx file in a Dokan mounted drive

### DISCUSSION THREAD

**Discussion by Thomas-Giaroli** [dokan-dev/dokan-dotnet#332] [Q&A] [answered]
Hello, i have a drive mounted on the R: letter that i created with Dokan, the same piece of software also communicate with a distant server asking for files, and i use Dokan to represent these files on the R: drive. I have no problem opening any of these files but one, a docx file. When i double click it it will always open MS Word asking me to choose an encoding method for another file, "mail.MSG", which is also on my dokan mounted drive. I seem to have no problem opening the docx file through other means like opening it through Word's interface or even online with Office 365. I have tried a lot of things and looked for similar issue but didn't find anything that could help me. 

Here is a gist containing the most important pieces of code in my dokan-dotnet based software, let me know if you need anything else : 

https://gist.github.com/Thomas-Giaroli/e446313ca234a6e00376de507908477d

Thanks already for any help you might be able to provide me,
Thomas Giaroli

**Comment by Liryna** [comment]
Hi @Thomas-Giaroli , this is really weird. I looked at the gist but could not see anything special.
After asking you to choose the encoding, does it still open the right file ?
Have you looked with procmon the activity of MS Word at that moment ?

**Reply by Thomas-Giaroli** [reply]
Hi Liryna, i just tried and no, it still tries to open the .MSG file after i chose an encoding method. I tried looking at Procmon but i don't seem to see anything, but that may be because i'm inexperienced with Procmon and Word's normal behaviour. I will link the capture i did if you want to take a look at it, by the way i can't directly upload .PML files here so if you'd like this more than the .CSV file let me know and i'll upload it on another platform. Let me know if you need anything else to determine the cause of the problem.

[Logfile.CSV](https://github.com/dokan-dev/dokan-dotnet/files/11540324/Logfile.CSV)

**Comment by Olof-Lagerkvist** [comment]
There seems to be nothing in that log that indicates that Word actually asks for any .MSG file. My best guess for now is that you might have an issue in your Dokan implementation that returns data for another file than the file that Word requests and that it causes Word to get file data in a format that it does not know and therefore asks for encoding options. Are you really sure about what file names your Dokan service receives requests for? There are several different possible behaviours, such as file name supplied in CreateFile method and otherwise file name supplied directly in ReadFile etc methods but instead no open file handle at that point.

**Reply by Thomas-Giaroli** [reply]
That effectively could be the problem, but Word is really trying to open the .MSG file, at least that's what he is displaying, he actually displays the name of the .MSG file, so i don't think he's just getting the wrong data, i really might be wrong though but that's my current opinion on this problem. I mean, i don't get how Word could display the name of the .MSG file if he was trying to open the .docx one with the wrong data.

![image](https://github.com/dokan-dev/dokan-dotnet/assets/72007344/715e1f75-d052-4640-b44d-fa5fcfff95fa)

I'll mostly be working on debugging this today so I will keep you updated if i ever find anything, thanks already for your suggestions!

**Reply by Olof-Lagerkvist** [reply]
Okay, yes, then the problem is more probably something else. I wonder if it could instead be a problem in the find files implementation. Could it be that Explorer asks for information about a file that the user double clicked, but you mistakenly return listing of all files in the directory and this MSG file just happen to come first?

**Reply by Thomas-Giaroli** [reply]
I think you might be right! My FindFiles implementation by default return all the files i get from my distant server, so that's already a problem. The thing is that, using VS debugger/spies, i saw that fileName in the findFiles function is always equal to "\\", never something like "\\mail.msg" or "\\wordFile.docx", I'm not really familiar with Dokan so i don't know, but is this normal behaviour? Or should i get a path to a file when trying to open any file on my mounted drive?

**Reply by Olof-Lagerkvist** [reply, marked as answer]
You should filter out what files you return in the file listing by the search pattern that the function is called with. Otherwise you will get exactly this problem.

**Reply by Thomas-Giaroli** [reply]
You were right, implementing FindFilesWithPattern solved my problem, thanks a lot!
